### PR TITLE
Make HealthEndpoint compatible with ALV

### LINKED ISSUE #877: Make HealthEndpoint compatible with latest SpringBoot version
The latest versions of Spring Boot actuators have a newer schema application/vnd.spring-boot.actuator.v3+json which Steeltoe needs to support. 
New Functionality Needs to be backwards compatible and hidden via feature toggle exposed by configuration.

### PR DESCRIPTION
**Pull request by hananiel**
## Description
Make health actuator response compatible with ALV. 

Fixes #877

## Quality checklist

- [ ] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [ ] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs [hunk 1]
```diff
@@ -8,87 +8,91 @@
...
+                }
+                else
+                {
+                    options = GetSerializerOptions(null);
+                }
+
+                return JsonSerializer.Serialize(result, options);
             }
-            else
+            catch (Exception e)
```
Comment (TimHess) [inline src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs:65]: Should we catch something more specific and potentially handle the [potentially thrown types](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serialize?view=net-6.0) differently? Is there potential for `NotSupportedException` or do we definitely always know all the types being serialized?

File: src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs [hunk 2]
```diff
@@ -8,87 +8,91 @@
...
     protected IEndpoint<TResult> endpoint;
     protected ILogger logger;
     protected IManagementOptions managementOptions;
 
-    public EndpointMiddleware(IManagementOptions managementOptions, ILogger logger = null)
-    {
-        this.logger = logger;
-        this.managementOptions = managementOptions ?? throw new ArgumentNullException(nameof(managementOptions));
-        if (this.managementOptions is ManagementEndpointOptions options)
+        public EndpointMiddleware(IManagementOptions managementOptions, ILogger logger = null)
```
Comment (TimHess) [inline src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs:20]: Was the whitespace change in this file intentional?

File: src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs (hunk 1 above)
Comment (hananiel) [reply src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs]: Good idea. Made tha change.

File: src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs (hunk 2 above)
Comment (hananiel) [reply src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs]: No it was the result of a bad merge.

**Review by TimHess — APPROVED** [review]
LGTM

File: src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs [hunk 3]
```diff
@@ -73,11 +72,16 @@ public virtual string Serialize(TResult result)
...
         serializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
 
-        if (serializerOptions.Converters?.Any(c => c is HealthConverter) != true)
+        if (serializerOptions.Converters?.Any(c => c is JsonStringEnumConverter) != true)
+        {
+            serializerOptions.Converters.Add(new JsonStringEnumConverter());
+        }
+
+        if (serializerOptions.Converters?.Any(c => c is HealthConverter or HealthConverterV3) != true)
+
```
Comment (TimHess) [inline src/Management/src/EndpointBase/Middleware/EndpointMiddleware.cs:84]:
```suggestion
```

**Review by TimHess — APPROVED** [review]

**Pull request merged**